skyportal/handlers/api/shift.py

```diff
@@ -141,86 +140,95 @@ def get(self, group_id=None):
-                    self.current_user,
-                    options=[
-                        joinedload(CommentOnShift.author),
-                        joinedload(CommentOnShift.groups),
-                    ],
+            shifts = []
+            for shift in queried_shifts:
+                susers = []
+                for su in shift.shift_users:
+                    user = su.user.to_dict()
+                    user["admin"] = su.admin
+                    user["needs_replacement"] = su.needs_replacement
+                    del user["oauth_uid"]
+                    susers.append(user)
+                gusers = []
+                for gu in shift.group.group_users:
+                    user = gu.user.to_dict()
+                    user["admin"] = gu.admin
+                    del user["oauth_uid"]
+                    gusers.append(user)
+
+                shift = shift.to_dict()
+                shift["shift_users"] = susers
+                shift["group"] = shift["group"].to_dict()
+                shift["group"]["group_users"] = gusers
+                comments = (
+                    session.scalars(
+                        CommentOnShift.select(
+                            self.current_user,
+                            options=[
+                                joinedload(CommentOnShift.author),
+                                joinedload(CommentOnShift.groups),
+                            ],
+                        ).where(CommentOnShift.shift_id == shift["id"])
+                    )
+                    .unique()
+                    .all()
                 )
-                .filter(CommentOnShift.shift_id == shift["id"])
-                .all()
-            )
-            shift["comments"] = sorted(
-                (
-                    {
-                        **{
-                            k: v
-                            for k, v in c.to_dict().items()
-                            if k != "attachment_bytes"
-                        },
-                        "author": {
-                            **c.author.to_dict(),
-                            "gravatar_url": c.author.gravatar_url,
-                        },
-                    }
-                    for c in comments
-                ),
-                key=lambda x: x["created_at"],
-                reverse=True,
-            )
-            shifts.append(shift)
+                shift["comments"] = sorted(
+                    (
+                        {
+                            **{
+                                k: v
+                                for k, v in c.to_dict().items()
+                                if k != "attachment_bytes"
+                            },
+                            "author": {
+                                **c.author.to_dict(),
+                                "gravatar_url": c.author.gravatar_url,
+                            },
+                        }
+                        for c in comments
```

Comment:
I know this is just the way it was, but I feel it would be more readable as a simple loop.

Reply:
I think if we really care, we can do the various instance of these in another PR.